cpp/include/games/nim/Game.hpp

```diff
@@ -116,40 +84,30 @@ struct Game {
     }
     static std::string compact_state_repr(const State& state) {
       std::ostringstream ss;
-      ss << "[" << state.stones_left << ", " << state.current_player << "]";
+      ss << "[" << state.stones_left << ", " << state.current_player << ", " << state.chance_active
+         << "]";
```

Comment:
In my opinion, a string like `"p0@13"` or `"p1@10*"` would be better. The `p0` indicates "player 0", while `*` indicates chance active.